### ISSUE TITLE
RequestParser.add_argument location with 'headers' in list format

### ISSUE THREAD

**Issue by strattonw** [flask-restful/flask-restful#527]
I have the code below which works just fine
```python
parser_ping = reqparse.RequestParser()
parser_ping.add_argument('X-ACCESS-KEY', type=str, help="X-ACCESS-KEY is required", required=True, location='headers')
parser_ping.add_argument('source')
parser_ping.add_argument('message')

...

args = parser_ping.parse_args()
```
but if I alter the code to
```python
parser_ping = reqparse.RequestParser()
parser_ping.add_argument('X-ACCESS-KEY', type=str, help="X-ACCESS-KEY is required", required=True, location=['headers'])
parser_ping.add_argument('source')
parser_ping.add_argument('message')

...

args = parser_ping.parse_args()
```
where `location=[...]` any length list, arguments are not being parsed from HTTP Headers and are always returning `"X-ACCESS-KEY": "(X-ACCESS-KEY is required)  Missing required parameter in the HTTP headers"`.

**Comment by joshfriend** [comment]
Ok, so this is a fun one. `Request.headers` returns an [`EnvironHeaders`][environheaders] object, which inherits from `ImmutableHeadersMixin` and `Headers`. The `Headers` class operates similarly to a [`MultiDict`][multidict] (can have multiple values for a single key), but does not actually inherit from `MultiDict`. This is because the [keys are not case sensitive][headers-case] as per [RFC2616 section 4.2][rfc2616-sec4].

When multiple locations are specified for an argument, the request parser collects all the arguments from all the locations [into a single MultiDict][rp-combine]. This removes the case insensitivity of the headers and translates the keys to their [title case equivalents][title-case].

tl;dr: If you use `parser.add_argument('X-Access-Key', location=['headers'])`, it will work. However, for the sake of robustness, use `location='headers'` since that will return the `Headers` object directly to the request parser and will be case insensitive.

This feels strange to me, but I can't think of a better alternative at the moment, so I suppose a note in the docs should suffice for now.

[environheaders]: https://github.com/mitsuhiko/werkzeug/blob/0.10.4/werkzeug/datastructures.py#L1255
[multidict]: https://github.com/mitsuhiko/werkzeug/blob/0.10.4/werkzeug/datastructures.py#L318
[headers-case]: https://github.com/mitsuhiko/werkzeug/blob/0.10.4/werkzeug/datastructures.py#L891-L894
[rfc2616-sec4.2]: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
[rp-combine]: https://github.com/flask-restful/flask-restful/blob/0.3.4/flask_restful/reqparse.py#L96
[title-case]: https://github.com/mitsuhiko/werkzeug/blob/0.10.4/werkzeug/datastructures.py#L1289
